packages/fuels-core/src/types/wrappers/transaction.rs

```diff
@@ -190,42 +157,21 @@ impl Transaction for TransactionType {
         }
     }
 
-    fn with_gas_price(self, gas_price: u64) -> Self {
```

Comment:
Why was this removed?

Reply:
I thought it would change the tx id so I did not want to user to think that he can just use `with_gas_price` after the tx was built and the ID was set. 

However, I was wrong! :D these methods do not change the id so I have returned all of them (maturity, gas_price, gas_limit )